### ISSUE TITLE
"zip bomb detected" error when uploading a spreadsheet

### ISSUE THREAD

**Issue by nightowlaz** [IQSS/dataverse#7854]
hi there .. we had a depositor try to upload a spreadsheet into his dataset today and the ingest failed (which makes sense as the spreadsheet has multiple tabs and graphs included), but the error that showed on the ingest error was a little bit alarming ... the actual error on the screen and in the log was:
```
"[#|2021-05-06T16:35:24.362+0000|WARNING|Payara 5.2020.6|edu.harvard.iq.dataverse.ingest.IngestServiceBean|_ThreadID=1346;_ThreadName=orb-thread-pool-1 (pool #1): worker-2;_TimeMillis=1620318924362;_LevelValue=900;|
  Ingest failure (IO Exception): Could not parse Excel/XLSX spreadsheet. Zip bomb detected! The file would exceed the max. ratio of compressed file size to the size of the expanded data.
This may indicate that the file is used to inflate memory usage and thus could pose a security risk.
You can adjust this limit via ZipSecureFile.setMinInflateRatio() if you need to work with files which exceed this limit.
Uncompressed size: 103141, Raw/compressed size: 724, ratio: 0.007020
Limits: MIN_INFLATE_RATIO: 0.010000, Entry: xl/media/image2.emf.|#]"
```
I believe the error generates from Apache POI, but the entire error above posts on the site ..
<img width="334" alt="zip-bomb-error" src="https://user-images.githubusercontent.com/22484650/117367402-fc778f80-ae76-11eb-8d0f-d824769dd12d.png">

I posted it on chat and it was suggested that I create an issue for the possibility of making code changes to display a different error to the user. Thank you!

**Comment by pdurbin** [comment]
@nightowlaz yes, thanks for opening this issue and chatting yesterday at http://irclog.iq.harvard.edu/dataverse/2021-05-06#i_137436

https://stackoverflow.com/questions/44897500/using-apache-poi-zip-bomb-detected seems to indicate that we might be able to do something about how we're using Apache POI to parse Excel files. (If you are able to share the Excel file you were testing, it would be helpful.) I'm sure in most cases, the Excel file is not actually a zip bomb. 💣

**Comment by nightowlaz** [comment]
oh yes, sorry, meant to attach that, too ... here is the file that caused the error (renamed).
[asu-testfile0507.xlsx](https://github.com/IQSS/dataverse/files/6442822/asu-testfile0507.xlsx)

**Comment by pdurbin** [comment]
I don't think upgrading will make a difference for you. I'm not sure why the error looks different on demo. No the versions are slightly different. The "develop" branch is always a bit ahead of what we have on the demo site, which is the latest released version.

**Comment by pdurbin** [comment]
@nightowlaz it looks like you're on the latest, 5.11.1. Are you still getting this odd error? Thanks.

**Comment by qqmyers** [comment]
FWIW: I hit this in another applications and set `org.apache.poi.openxml4j.util.ZipSecureFile.setMinInflateRatio(0.001);` before calling POI. That could be done in Dataverse as well as there are some legitimate files that compress better than the default 0.01 ratio. (The downside is that it malicious users also get a factor of 10 in being able to send smaller zips that end up taking more space/ it takes longer to detect real Zip bombs that expand by huge factors.) A workaround would probably be to rezip with compression turned off or limited for affected files.

**Comment by nightowlaz** [comment]
We haven't seen this error anymore. Looking back at our notes, the file that we were getting it on was a spreadsheet that was not in any organized format (ie: not in the required columns with titles, but data all over the place), and I think we ended up setting the tabular ingest size limit, and upgraded, and don't seem to see it any more. We talked about changing the MinInflateRatio but don't think we ever did.
